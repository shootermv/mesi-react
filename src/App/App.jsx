import React from 'react';
import { Router, Route } from 'react-router-dom';
import { connect } from 'react-redux';

import { history } from '../_helpers';
import { alertActions } from '../_actions';
import { PrivateRoute } from '../_components';
import { HomePage } from '../HomePage';
import { AdminPage } from '../AdminPage';
import { LoginPage } from '../LoginPage';
import { RegisterPage } from '../RegisterPage';
import { PrivatePage } from '../PrivatePage';

import { AppHeader } from '../AppHeader';


export class App extends React.Component {
    constructor(props) {
        super(props);

        const { dispatch } = this.props;
        history.listen((location, action) => {
            // clear alert on location change
            dispatch(alertActions.clear());
        });

    }

    render() {
        const { alert } = this.props;
        return (
            <div className="jumbotron">

                <AppHeader/>

                <div className="container">
                    <div className="col-sm-8 col-sm-offset-2">
                        {alert.message &&
                            <div className={`alert ${alert.type}`}>{alert.message}</div>
                        }
                        <Router history={history}>
                            <div>
                                <PrivateRoute exact path="/" component={HomePage} />
                                <PrivateRoute exact path="/private" component={PrivatePage} />
                                <PrivateRoute exact path="/admin" component={AdminPage} />
                                <Route path="/login" component={LoginPage} />
                                <Route path="/register" component={RegisterPage} />
                            </div>
                        </Router>
                    </div>
                </div>
            </div>
        );
    }
}

function mapStateToProps(state) {
    const { alert } = state;
    return {
        alert
    };
}


export default connect(mapStateToProps)(App); 