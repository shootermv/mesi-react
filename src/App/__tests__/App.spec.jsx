// test file

import React from 'react';
import {shallow} from 'enzyme'


import {App} from '../App'

// var wrapper = enzyme.shallow(<App />);
test('Should render AdminPage component', () => {
    const alert = {
        messsage: '',
        type: ''
    }
    const wrapper = shallow(<App alert={alert}/>);
    expect(wrapper).toMatchSnapshot();
});