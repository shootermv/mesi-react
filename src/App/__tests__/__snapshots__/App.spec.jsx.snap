// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render AdminPage component 1`] = `
<div
  className="jumbotron"
>
  <AppHeader />
  <div
    className="container"
  >
    <div
      className="col-sm-8 col-sm-offset-2"
    >
      <Router
        history={
          Object {
            "action": "POP",
            "block": [Function],
            "createHref": [Function],
            "go": [Function],
            "goBack": [Function],
            "goForward": [Function],
            "length": 1,
            "listen": [Function],
            "location": Object {
              "hash": "",
              "pathname": "/",
              "search": "",
              "state": undefined,
            },
            "push": [Function],
            "replace": [Function],
          }
        }
      >
        <div>
          <PrivateRoute
            component={[Function]}
            exact={true}
            path="/"
          />
          <PrivateRoute
            component={[Function]}
            exact={true}
            path="/private"
          />
          <PrivateRoute
            component={[Function]}
            exact={true}
            path="/admin"
          />
          <Route
            component={[Function]}
            path="/login"
          />
          <Route
            component={[Function]}
            path="/register"
          />
        </div>
      </Router>
    </div>
  </div>
</div>
`;
